Set an $ORIGIN rpath for the Linux editor

mikix opened the pull request:
This lets users skip the LD_LIBRARY_PATH workaround script.

### What is $ORIGIN

If a Linux binary (or library) has an rpath that includes `$ORIGIN`, it will look in its own directory for linked libraries. (i.e. it will act as if `LD_LIBRARY_PATH=$MYDIR` was set. All automatically!

Here's an [article](https://nehckl0.medium.com/creating-relocatable-linux-executables-by-setting-rpath-with-origin-45de573a2e98) about it.

### This PR

By setting this rpath value, you should be able to drop the Linux caveat warnings about finding libsteam_api.so and also basically drop the `godotsteam.sh` wrapper.

That said, I did not change any documentation around Linux caveats yet (I imagine a similar change should happen in the plugin branches as well first, plus maybe a release or two to give folks times to upgrade).

I also didn't remove the `godotsteam.sh` wrapper, since maybe users are used to it. But I did turn it into a just a tiny passthrough script to the editor binary (all arguments to it will get passed along).

### Testing

I tested these commands separately locally, but I did not actually test-run the CI pipeline -- I'm not super familiar with this project.

Gramps commented:
That's pretty neat!  I definitely do not mind getting rid of the godotsteam.sh file at all.  I'm still not 100% why it affects some users and not others.

One thing, this issue still comes up with exported projects run outside of the Steam client.  Would applying this to templates also help?

mikix commented:
> Would applying this to templates also help?

Probably…? I’m not super familiar with how godot templates work, but this trick can be used on any executable or library.

I can test when I’m back at my computer

mikix commented:
Aha. I think I have a more elegant fix, and I also understand why it might have appeared inconsistent for users -- looks like the current scons script is setting an RPATH of `.` -- which means it will only look in whatever the current working directory is. So if the user is ACTIVELY in a directory that has `libsteam_api.so` it will work. But if they are in some other directory, it won't work.

But `$ORIGIN` is like `.` but relative to the executable itself. So that's what we want instead.

I also tested and it fixes the templates too. I'll update this PR in a bit.

mikix commented:
OK I updated this PR:
- Dropped `godotsteam.sh` entirely
- Updated `SCsub` to use a better RPATH setting (the weird escaping I stole from the [scons wiki](https://github.com/SCons/scons/wiki/UsingOrigin))
- Dropped a reference to the issue from the README

I'll work on versions for godot4 and the plugin branches.

mikix commented:
Interestingly, the gdextension version seems to not need this -- looks like the 4.0 godot-cpp [sets $ORIGIN](https://github.com/search?q=repo%3Agodotengine%2Fgodot-cpp%20ORIGIN&type=code) for us.

Gramps commented:
Very excellent!  Thanks for all this!  My desktop always worked without the godotsteam.sh fix but my laptop always needed it, oddly.
